src/wallet/walletdb.cpp

```diff
@@ -810,6 +810,7 @@ void ThreadFlushWalletDB(const string& strFile)
                 if (nRefCount == 0)
                 {
                     boost::this_thread::interruption_point();
+                    const std::string& strFile = pwalletMain->strWalletFile;
```

Comment:
wallet.h says "This lock protects all the fields added by CWallet except for: ... strWalletFile (immutable after instantiation)"